lib/checkother.cpp

```diff
@@ -4384,6 +4384,41 @@ void CheckOther::overlappingWriteFunction(const Token *tok, const std::string& f
     reportError(tok, Severity::error, "overlappingWriteFunction", "Overlapping read/write in " + funcname + "() is undefined behavior");
 }
 
+void CheckOther::checkSuspiciousComma()
+{
+    if (!mSettings->severity.isEnabled(Severity::style)) {
+        return;
+    }
+
+    logChecker("CheckOther::warnSuspiciousComma");
+
+    for (const Token* tok = mTokenizer->list.front(); tok; tok = tok->next()) {
+        if (tok->str() == "," && tok->isBinaryOp()) {
+            const Token * parent = tok->astParent();
+            if (parent && Token::Match(parent->previous(), "if|while (")) {
+                if (tok->strAt(-1) == ")") {
+                    const Function * func = tok->linkAt(-1)->previous()->function();
+                    if (func && func->initArgCount > 0) {
+                        const Token * r_op = tok->astOperand2();
+                        if (r_op && (r_op->isVariable() ||
+                                     r_op->tokType() == Token::eNumber ||
+                                     r_op->tokType() == Token::eString ||
+                                     r_op->tokType() == Token::eChar ||
+                                     r_op->tokType() == Token::eBoolean)) {
+                            checkSuspiciousCommaError(tok);
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+void CheckOther::checkSuspiciousCommaError(const Token *tok)
+{
+    reportError(tok, Severity::style, "warnSuspiciousComma", "There is an suspicious comma expression used as a condition in an if/while condition statement.");
```

Comment:
Typo: a suspicious
I would shorten the message to e.g. "There is a suspicious comma expression in an if/while condition statement." and change the ID to just "suspiciousComma"